pg_sparse/README.md

```diff
@@ -0,0 +1,133 @@
+# pg_sparse
+
+## Overview
+
+`pg_sparse` is a PostgreSQL extension that enables similarity search over sparse vectors
+in Postgres using HNSW.
+
+## Installation
+
+### From Self-Hosted Postgres
+
+If you are self-hosting Postgres and would like to use the extension within your existing
+Postgres, follow these steps:
+
+1. Install Rust and cargo-pgrx:
+
+```bash
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+cargo install cargo-pgrx --version 0.9.8
+```
+
+2. Then, run:
+
+```bash
+# Clone the repo (optionally pick a specific version)
+git clone https://github.com/paradedb/paradedb.git --tag <VERSION>
+
+# Install pg_sparse
+cd pg_sparse/
+cargo pgrx init --pg<YOUR-POSTGRES-MAJOR_VERSION>=`which pg_config`
+cargo pgrx install
+```
+
+You can then create the extension in your database by running:
+
+```sql
+CREATE EXTENSION pg_sparse;
+```
+
+If you are using a managed Postgres service like Amazon RDS, you will not be able to install `pg_sparse` until the Postgres service explicitly supports it.
+
+## Usage
+
+### Indexing
+
+By default, the `pg_sparse` extension creates a table called `paradedb.mock_items`
+that you can use for quick experimentation.
+
+To index a table, use the following SQL command:
+
+```sql
+CREATE INDEX idx_mock_items
+ON paradedb.mock_items
+USING sparse_hnsw(sparse_embedding)
+```
+
+TODO: Implement index creation, insertion, and scans.
+
+## Development
+
+### Prerequisites
+
+Before developing the extension, ensure that you have Rust installed
+(version >1.70), ideally via `rustup` (we've observed issues with installing Rust
+via Homebrew on macOS).
+
+Then, install and initialize pgrx:
+
+```bash
+cargo install cargo-pgrx --version 0.9.8
```

Comment:
Just making a note to update this pgrx too, it'll need to match what we have in our repo.